File: 2_Programação_sistemas_operacionais/2.09_Sockets/01_cliente_local.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <sys/socket.h>
#include <sys/un.h>
#include <unistd.h>

int main (int argc, char* const argv[])
{
	if(argc < 3)
	{
		puts("Modo de Uso:");
		printf("%s NOME_SOCKET MSG\n", argv[0]);
		exit(1);
	}
	char *socket_name;
	char *mensagem;
	int socket_id;
	struct sockaddr name;
	int length;

	socket_name = argv[1];
	mensagem = argv[2];
	socket_id = socket(PF_LOCAL, SOCK_STREAM,0);
	
	name.sa_family = AF_LOCAL;
	strcpy(name.sa_data, socket_name);
	connect(socket_id, &name, sizeof(name));
	length = strlen(mensagem) + 1;
	write(socket_id, &length, sizeof(int));
	write(socket_id, mensagem, length);
	close(socket_id);
	return 0;
}

File: 2_Programação_sistemas_operacionais/2.7_Sockets/Python/01_cliente_local.py
import socket, struct, sys

if len(sys.argv) < 3:
	print("Modo de Uso:")
	print("%s NOME_SOCKET MSG" % sys.argv[0])
	exit(1)

socket_name = sys.argv[1]
mensagem    = sys.argv[2]

socket_id = socket.socket(socket.AF_UNIX, socket.SOCK_STREAM)
socket_id.connect("\0" + socket_name)
length = int(len(mensagem))
socket_id.send(struct.pack('i',length))
socket_id.send(mensagem.encode('utf-8'))
socket_id.close()